nornir/core/inventory.py

```diff
@@ -196,42 +196,52 @@ def nornir(self, value):
             self._nornir = value
 
     @property
-    def host(self):
-        """String used to connect to the device. Either ``nornir_host`` or ``self.name``"""
-        return self.get("nornir_host", self.name)
+    def hostname(self):
+        """String used to connect to the device. Either ``hostname`` or ``self.name``"""
+        return self.get("hostname", self.name)
+
+    @property
+    def port(self):
+        """Either ``port`` or ``None``."""
+        return self.get("port")
 
     @property
     def username(self):
         """Either ``nornir_username`` or user running the script."""
-        return self.get("nornir_username", getpass.getuser())
+        return self.get("username", getpass.getuser())
 
     @property
     def password(self):
         """Either ``nornir_password`` or empty string."""
-        return self.get("nornir_password", "")
-
-    @property
-    def ssh_port(self):
-        """Either ``nornir_ssh_port`` or ``None``."""
-        return self.get("nornir_ssh_port")
+        return self.get("password", "")
 
     @property
-    def network_api_port(self):
-        """
-        For network equipment this is the port where the device's API is listening to.
-        Either ``nornir_network_api_port`` or ``None``.
-        """
-        return self.get("nornir_network_api_port")
-
-    @property
-    def os(self):
-        """OS the device is running. Defaults to ``nornir_os``."""
-        return self.get("nornir_os")
-
-    @property
-    def nos(self):
-        """Network OS the device is running. Defaults to ``nornir_nos``."""
-        return self.get("nornir_nos")
+    def platform(self):
+        """OS the device is running. Defaults to ``platform``."""
+        return self.get("platform")
+
+    def get_connection_parameters(
+        self, connection: Optional[str] = None
+    ) -> Dict[str, Any]:
+        if not connection:
+            return {
+                "hostname": self.hostname,
+                "port": self.port,
+                "username": self.username,
+                "password": self.password,
+                "platform": self.platform,
+                "connection_options": {},
+            }
+        else:
+            conn_params = self.get("connection_options", {}).get(connection, {})
+            return {
+                "hostname": conn_params.get("hostname", self.hostname),
+                "port": conn_params.get("port", self.port),
+                "username": conn_params.get("username", self.username),
+                "password": conn_params.get("password", self.password),
+                "platform": conn_params.get("platform", self.platform),
+                "connection_options": conn_params.get("additional_options", {}),
+            }
 
```

Comment:
I was thinking about this and I have mixed feelings. My concern is that it is a little bit too much magic which can be not so intuitive, but I also understand the reasoning behind including this.
What I think could be less magic:
To form a dictionary with default Host attributes `hostname`, `port`, `username`, `password`, `platform` and then just update with `connection_options`. `connection_options` would always be provided in a format understandable by the backend library (for the plugins we are shipping with nornir, user-defined plugins can do anything). For example, if the backend library uses `pass` instead of `password`, if you want to override host `password`, in `connection_options` you must define `pass`. Then to pass one resulting dictionary to the plugin and let it form its own args. In the connect plugin it would be something like:
`conn_params['pass'] = options.get('pass', options.get('password'))`
This looks a little ugly, but it feels a little more explicit.

I don't know what is best here, 50/50.